src/Vm.sol

```diff
@@ -113,30 +121,63 @@ interface VmSafe {
-    // Path is relative to the project root.
+    // `path` is relative to the project root.
     function writeFileBinary(string calldata path, bytes calldata data) external;
     // Writes line to file, creating a file if it does not exist.
+    // `path` is relative to the project root.
     function writeLine(string calldata path, string calldata data) external;
     // Closes file for reading, resetting the offset and allowing to read it from beginning with readLine.
+    // `path` is relative to the project root.
     function closeFile(string calldata path) external;
-    // Removes file. This cheatcode will revert in the following situations, but is not limited to just these cases:
-    // - Path points to a directory.
+    // Removes a file from the filesystem.
+    // This cheatcode will revert in the following situations, but is not limited to just these cases:
+    // - `path` points to a directory.
     // - The file doesn't exist.
     // - The user lacks permissions to remove the file.
+    // `path` is relative to the project root.
     function removeFile(string calldata path) external;
+    // Creates a new, empty directory at the provided path.
+    // This cheatcode will revert in the following situations, but is not limited to just these cases:
+    // - User lacks permissions to modify `path`.
+    // - A parent of the given path doesn't exist and `recursive` is false.
+    // - `path` already exists and `recursive` is false.
+    // `path` is relative to the project root.
+    function createDir(string calldata path, bool recursive) external;
+    // Removes a directory at the provided path.
+    // This cheatcode will revert in the following situations, but is not limited to just these cases:
+    // - `path` doesn't exist.
+    // - `path` isn't a directory.
+    // - User lacks permissions to modify `path`.
+    // - The directory is not empty and `recursive` is false.
+    // `path` is relative to the project root.
+    function removeDir(string calldata, bool recursive) external;
+    // Reads the directory at the given path recursively, up to `max_depth`.
+    // `max_depth` defaults to 1, meaning only the direct children of the given directory will be returned.
+    // Follows symbolic links if `follow_links` is true.
+    function readDir(string calldata path) external returns (DirEntry[] memory entries);
+    function readDir(string calldata, uint64 maxDepth) external returns (DirEntry[] memory entries);
+    function readDir(string calldata, uint64 maxDepth, bool followLinks) external returns (DirEntry[] memory entries);
+    // Reads a symbolic link, returning the path that the link points to.
+    // This cheatcode will revert in the following situations, but is not limited to just these cases:
+    // - `path` is not a symbolic link.
+    // - `path` does not exist.
+    function readLink(string calldata linkPath) external returns (string memory targetPath);
```

Comment:
Should these be `read`? `fsMetadata` currently isn't, but `readFile` is.

Reply:
You mean `view` I assume? The heuristic we use for determining view/pure is here: https://github.com/foundry-rs/forge-std/pull/205#issuecomment-1301096547. Based on that, mkaing all fs-related cheats that just read data and make no state changes `view` seems like the way to go